### PR TITLE
Reduce packaging test log output and introduce ready request interval

### PR DESCRIPTION
**Pull request by mark-vieira**
As [discussed here](https://github.com/elastic/infra/issues/15385#issuecomment-544642362) this PR reduces the log output produced by packaging test while the tests wait for the ES cluster to start. It also introduces an interval in between checks to see if the cluster is ready so we aren't effectively DoSing our cluster during test startup.

### CODE REVIEW

**Comment by atorok** [comment]
Would it be overkill to create a small subset from build-tools that would be added as a dependency to tests ?  We could split out a base package from test:framework that has no dependencies and have it live in buildSrc that would solve other similar issues too because we could add it to maven central and use it in test framework too. 
We have code in the build that does something similar for test clusters, and I bet we have some other places to ...

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging

File: qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java [hunk 1]
```diff
@@ -50,27 +51,33 @@ public static void waitForElasticsearch(String status, String index, Installatio
...
+                        throw new RuntimeException("Connecting to elasticsearch cluster health API failed:\n" + statusLine + "\n" + body);
+                    }
 
-                started = true;
+                    started = true;
 
-            } catch (IOException e) {
-                logger.info("Got exception when waiting for cluster health", e);
+                } catch (IOException e) {
+                    lastException = e;
```
Comment (rjernst) [inline qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: Instead of losing these completely, perhaps we should log at debug level?

**Review by rjernst — APPROVED** [review]
Looks fine, but I don't think we should swallow these exceptions completely.

File: qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java (hunk 1 above)
Comment (mark-vieira) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: If we logged at debug, what would we have to do to get at that log output? Looks like right now, tests log only at `INFO` and above.

Comment (rjernst) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: Then let's suppress them? Take the first exception, suppress additional exceptions, and we will get all of them if we fail.

Comment (mark-vieira) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: I'm not sure I follow. What we are doing right now is suppressing all exceptions except for the most recent one before reaching the timeout. Are you suggesting we do the same except that we capture only the _first_ exception we encounter?

Comment (rjernst) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: I mean actually "suppressed exceptions":

```
IOException foundException;
...
} catch {IOException e) {
    if (foundException == null) {
        foundException = e;
    } else {
        foundException.addSuppressed(e);
    }   
}
```

Comment (mark-vieira) [reply qa/os/src/test/java/org/elasticsearch/packaging/util/ServerUtils.java]: Ah. Honestly, I've never used suppressed exceptions before. Thanks for pointing this out. I've implemented as you described.

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Pull request merged**